Source/WebCore/rendering/RenderElement.cpp

```diff
@@ -1922,7 +1922,7 @@ bool RenderElement::checkForRepaintDuringLayout() const
 {
     if (document().view()->layoutContext().needsFullRepaint() || !everHadLayout() || hasSelfPaintingLayer())
         return false;
-    return !settings().repaintOutsideLayoutEnabled();
+    return true;
```

Comment:
or you could just 
return everHadLayout()  && !document().view()->layoutContext().needsFullRepaint() && !hasSelfPaintingLayer()

but either way is fine.